snuba/workers/tasks.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from typing import Any, Dict, Mapping, MutableMapping
+
+from snuba import settings
+from snuba.clickhouse.errors import ClickhouseError
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.storage import StorageNotAvailable
+from snuba.query.allocation_policies import AllocationPolicyViolations
+from snuba.query.exceptions import QueryPlanException
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request.schema import RequestSchema
+from snuba.request.validation import build_request, parse_snql_query
+from snuba.state.rate_limit import RateLimitExceeded
+from snuba.utils.metrics.timer import Timer
+from snuba.web import QueryException, QueryTooLongException
+from snuba.web.constants import get_http_status_for_clickhouse_error
+from snuba.web.query import parse_and_run_query
+from snuba.workers import celery_app
+
+logger = logging.getLogger("snuba.tasks")
+
+
+@celery_app.task(serializer="pickle")
+def send_query(
+    dataset: Dataset, body: Dict[str, Any], timer: Timer, referrer: str
+) -> Mapping[str, Any]:
+    schema = RequestSchema.build(HTTPQuerySettings)
+    request = build_request(
```

Comment:
Since this is happening outside the normal flask app, all the handlers we have to catch exceptions and format them properly are not going to work. I think you will have to add something similar to ensure that the results stored in redis are a consistent format.